### PR TITLE
feat(replay): Use `<ObjectInspector>` for DOM Mutations

### PR DESCRIPTION
**Pull request by billyvg**
Also refactor to use a hook `useVirtualizedInspector` (I can split this out into a new PR).

image

Requires https://github.com/getsentry/sentry/pull/47432
Requires https://github.com/getsentry/react-inspector/pull/4

### CODE REVIEW

**Comment by billyvg** [comment]
Low priority, but thoughts on this @ryan953? We can display the entire tree since it's collapsed.

-- also @Jesse-Box had expressed concerns with the theming of `<ObjectInspector>` globally and imo this default fits in so nicely 😭

File: static/app/views/replays/detail/console/consoleLogRow.tsx [hunk 1]
```diff
@@ -24,7 +24,8 @@ type Props = {
   onDimensionChange?: (
     index: number,
     path: string,
-    expandedState: Record<string, boolean>
+    expandedState: Record<string, boolean>,
+    event: MouseEvent<HTMLDivElement>
   ) => void;
```
Comment (ryan953) [inline static/app/views/replays/detail/console/consoleLogRow.tsx:29]: nit: this type could come from the new hook somehow. to save re-typing it each time

File: static/app/views/replays/detail/domMutations/domMutationRow.tsx [hunk 2]
```diff
@@ -48,6 +58,12 @@ function DomMutationRow({
     [handleMouseLeave, breadcrumb]
   );
 
+  const handleDimensionChange = useCallback(
+    (path, expandedState, e) =>
+      onDimensionChange && onDimensionChange(index, path, expandedState, e),
+    [index, onDimensionChange]
+  );
```
Comment (ryan953) [inline static/app/views/replays/detail/domMutations/domMutationRow.tsx:65]: nit: should this be called `handleExpand` or something.

there's two method signatures for `*DimensionChange` so im just thinking aloud about making it clearer which is which

**Review by ryan953 — COMMENTED** [review]
haven't run the code but it looks good. lost of the existing console code moved into the hook and re-used for dom mutations 👍 


I agree the theme is really good already and i hope not a blocker.

Also i like being able to expand the html nodes fully. But nodes might take up a lot of vertical space though, so it might be a good feature to auto-expand only a few levels, and then let people dig deeper when it matters. Can circle back to that and also look at req/resp details at the same time.. they also could have a couple levels open be default but don't need everything to be expanded.

**Comment by Jesse-Box** [comment]
> I agree the theme is really good already and i hope not a blocker.
Definitely not a blocker from my end.

I say we should put this behind a feature flag, so we can play around with it. As a separate task @vuluongj20 can support us with building a theme which is aligned with our prism theme.

**Comment by vuluongj20** [comment]
> > I agree the theme is really good already and i hope not a blocker.
> > Definitely not a blocker from my end.
> 
> I say we should put this behind a feature flag, so we can play around with it. As a separate task @vuluongj20 can support us with building a theme which is aligned with our prism theme.

@Jesse-Box Will look into how we can fit our Prism theme in here. The default theme is good enough for now — I like how it mirrors the Chrome inspector, making the code more familiar & easier to scan.

**Comment by billyvg** [comment]
@vuluongj20 I wouldn't be opposed to keeping it as the default because of the familiarity with Chrome inspector!

**Comment by billyvg** [comment]
Closing this for now as the inspector library is a bit buggy for DOM (I can't get the expandLevels working quite right).

**Pull request closed without merging**